ckan/views/dataset.py

```diff
@@ -83,14 +83,12 @@ def drill_down_url(alternative_url=None, **by):
     )
 
 
-def remove_field(package_type, key, value=None, replace=None):
+def remove_field(key, value=None, replace=None):
     return h.remove_url_param(
         key,
         value=value,
         replace=replace,
-        controller=u'dataset',
-        action=u'search',
-        alternative_url=package_type
+        alternative_url=h.url_for(u'dataset.search')
```

Comment:
It won't work for custom dataset types, as i understand. This is the reason, why there was `package_type` argument

Reply:
Thank you for your explanation.
In my opinion, the root cause is likely to be due to the / at the end of the url path, unlike the previous release version. (e.g. localhost/dataset "**/**" , localhost/group "**/**" ....)

1. localhost/dataset **/**
2.  select facet -> localhost/dataset **/** ?res_format=csv
3.  remove facet ->  localhost/dataset **/** ?res_format=csv
4.  h.remove_url_param(alternative_url='dataset') ->  localhost/dataset **/** dataset

I would like to ask you if you know the relevant history.